Firestore/Example/Tests/SpecTests/FSTSyncEngineTestDriver.mm

```diff
@@ -144,6 +143,13 @@ @implementation FSTSyncEngineTestDriver {
   std::unordered_map<User, NSMutableArray<FSTOutstandingWrite *> *, HashUser> _outstandingWrites;
   DocumentKeySet _expectedLimboDocuments;
 
+  /** A dictionary for tracking the listens on queries. */
+  std::unordered_map<FSTQuery *,
```

Comment:
Very optional: maybe something like
```cpp
namespace objc {
// Optionally, enable_if could check K is an Objective-C class,
// though it could be overkill
template <typename K, typename V>
using unordered_map = std::unordered_map<
  K,
  V,
  objc::Hash<K>,
  objc::EqualTo<K>;
}
// ...
objc::unordered_map<FSTQuery*, std::shared_ptr<QueryListener>> _queryListeners;
```